src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -759,6 +762,37 @@ class WorkspaceService(userInfo: UserInfo, dataSource: DataSource, workspaceDAO:
     }
   }
 
+  def abortSubmission(workspaceName: WorkspaceName, submissionId: String) = {
+    dataSource inTransaction { txn =>
+      withWorkspace(workspaceName, txn) { workspace =>
+        requireAccess(GCSAccessLevel.Write, workspace, txn) {
+          withSubmission(workspace, submissionId, txn) { submission =>
+            val aborts = submission.workflows.map( wf =>
+              Try(executionServiceDAO.abort(wf.workflowId, userInfo.authCookie)) match {
+                case Success(_) => Success(wf.workflowId)
+                //NOTE: Cromwell returns 403 Forbidden if you try to abort a workflow that's already in a terminal
+                //status. This is fine for our purposes, so we turn it into a Success.
+                case Failure(ure:UnsuccessfulResponseException) if ure.response.status == StatusCodes.Forbidden => Success(wf.workflowId)
```

Comment:
Are there any cases where you could get Forbidden for a workflow that's not in a terminal status?